server/athenian/api/controllers/miners/github/pull_request.py

```diff
@@ -281,3 +340,96 @@ def dtmin(first: Union[DT, float], second: Union[DT, float]) -> DT:
     if second != second:
         return first
     return min(first, second)
+
+
+class PullRequestListMiner(PullRequestTimesMiner):
+    """Collect various PR metadata for displaying PRs on the frontend."""
+
+    def __init__(self, prs: pd.DataFrame, reviews: pd.DataFrame, review_comments: pd.DataFrame,
+                 comments: pd.DataFrame, commits: pd.DataFrame):
+        """Initialize a new instance of `PullRequestListMiner`."""
+        super().__init__(prs, reviews, review_comments, comments, commits)
+        self._stages = set()
+        self._participants = {}
+
+    @property
+    def stages(self) -> Set[Stage]:
+        """Return the required PR stages."""
+        return self._stages
+
+    @stages.setter
+    def stages(self, value: Sequence[Stage]):
+        """Set the required PR stages."""
+        self._stages = set(value)
+
+    @property
+    def participants(self) -> Dict[ParticipationKind, Set[str]]:
+        """Return the required PR participants."""
+        return self._participants
+
+    @participants.setter
+    def participants(self, value: Mapping[ParticipationKind, Sequence[str]]):
+        """Set the required PR participants."""
+        self._participants = {k: set(v) for k, v in value.items()}
+
+    def _match_participants(self, yours: Mapping[ParticipationKind, Set[str]]) -> bool:
+        """Check the PR particpants for compatibility with self.participants.
+
+        :return: True whether the PR satisifes the particpants filter, otherwise False.
+        """
+        for k, v in self.participants:
+            if not yours[k].intersection(v):
+                return False
+        return True
+
+    def _compile(self, pr: MinedPullRequest) -> Optional[PullRequestListItem]:
```

Comment:
Okay, I think this is what you're mentioning in Slack, so matching the participants and the stage, right?

Reply:
Exactly. As you see, there is no filter for the participants to make an action in the specified time interval.

Actually, after having a call with Waren, I think that these participants are actually a good approximation of what the product initially wanted.

Reply:
> there is no filter for the participants to make an action in the specified time interval.

What do you mean? Because there's the filtering on participants. Ah! Maybe you mean a filtering on participants doing a specific action in a specific time interval? Okay I got it now 👍. These are actions done in those PRs that have been already filtered, but those actions are not time-bounded in that same time-interval. Did I understand correctly?

Reply:
Yep :heavy_check_mark: